### PR TITLE
Feature/keti 1362

### PR DESCRIPTION
**Pull request by arjunnayak**

### CODE REVIEW

File: model/src/test/java/org/cloudfoundry/identity/uaa/account/UserInfoResponseJsonTests.java [hunk 1]
```diff
@@ -49,6 +49,11 @@ private static void assertHardcodedValues(String json) {
                 hasItems(CoreMatchers.is("Val 21"), CoreMatchers.is("Val 22"))
         );
 
+        assertThat(
```
Comment (bsekar) [inline model/src/test/java/org/cloudfoundry/identity/uaa/account/UserInfoResponseJsonTests.java]: Remove this. Meant to be a manual test.

File: model/src/test/resources/org/cloudfoundry/identity/uaa/account/UserInfoResponseJsonTests.json [hunk 2]
```diff
@@ -19,7 +19,8 @@
     "Key 2": [
       "Val 21",
       "Val 22"
-    ]
+    ],
+    "Key 3": ["Val 31"]
```
Comment (bsekar) [inline model/src/test/resources/org/cloudfoundry/identity/uaa/account/UserInfoResponseJsonTests.json]: Remove also this. Same as above

File: scripts/cargo/uaa.yml [hunk 3]
```diff
@@ -55,4 +55,4 @@ login:
     0HOZbbHClXmGUjGs+GS+xC1FO/am2suCSYqNB9dyMXfOWiJ1+TLJk+o/YZt8vuxC
     KdcZYgl4l/L6PxJ982SRhc83ZW2dkAZI4M0/Ud3oePe84k8jm3A7EvH5wi5hvCkK
     RpuRBwn3Ei+jCRouxTbzKPsuCVB+1sNyxMTXzf0=
-    -----END CERTIFICATE-----
+    -----END CERTIFICATE-----
```
Comment (bsekar) [inline scripts/cargo/uaa.yml]: Remove all cert related changes

File: server.crt [hunk 4]
```diff
@@ -0,0 +1,31 @@
+-----BEGIN CERTIFICATE-----
```
Comment (bsekar) [inline server.crt]: Remove all cert related changes

File: server.key [hunk 5]
```diff
@@ -0,0 +1,51 @@
+-----BEGIN RSA PRIVATE KEY-----
```
Comment (bsekar) [inline server.key]: Remove all cert related changes

**Comment by bsekar** [comment]
Add a new IntegrationTest in `SamlLoginWithLocalIdpIT` that tests adding a custom attribute and seeing it federated correctly in the shadow account

**Comment by bsekar** [comment]
> Add a new IntegrationTest in `SamlLoginWithLocalIdpIT` that tests adding a custom attribute and seeing it federated correctly in the shadow account

Not applicable. Check [this comment](https://guidewirejira.atlassian.net/browse/KETI-1362?focusedCommentId=196945)

File: model/src/main/java/org/cloudfoundry/identity/uaa/scim/ScimUser.java [hunk 6]
```diff
@@ -842,4 +846,19 @@ public void patch(ScimUser patch) {
         }
     }
 
+    public static class JsonCustomAttributeSerializer extends JsonSerializer<LinkedHashMap<String, Object>> {
+        @Override
+        public void serialize(LinkedHashMap<String, Object> map, JsonGenerator gen,
+                SerializerProvider serializers) throws IOException {
+            gen.writeStartObject();
+            for(String e : map.keySet()) {
```
Comment (cwrm56) [inline model/src/main/java/org/cloudfoundry/identity/uaa/scim/ScimUser.java:854]: There's some inconsistency in formatting in the repo, but should this have space between for and paren?

**Review by cwrm56 — APPROVED** [review]
Looks good, couple of minor points, formatting with spaces missing in "for(" and what should we do if the object for the custom attribute is neither String nor List?  Should we log a warning, if nothing else?

**Pull request merged**